apps/smeargle/src/components/Filter/FilterDetailed/index.tsx

```diff
@@ -0,0 +1,28 @@
+import clsx from 'clsx';
+import { useFilterContext } from '../FilterProvider';
+import InstrumentSelector from './FilterDetailedFields/InstrumentSelector';
+import CategorySelector from './FilterDetailedFields/CategorySelector';
+import ZipCodeField from './FilterDetailedFields/ZipCodeField';
+import { useFilterDetailed } from './useFilterDetailed';
+import { Form } from 'solid-start/data/Form';
+
+export default function FilterDetailed() {
+  const [filterState] = useFilterContext();
+  const { submitFilter } = useFilterDetailed();
+  return (
+    <Form
+      onSubmit={submitFilter}
+      class={clsx(
+        'dark:bg-joplin border-1 dark:border-janis border-elvis dark:shadow-filter-dark shadow-filter-light bg-snow absolute flex h-20 w-fit translate-y-[55px] items-center rounded-full transition-all duration-300'
+      )}
+    >
+      <CategorySelector value={filterState.filter.category?.value ?? 'all'} />
+      <div class="mr-2" />
+      <InstrumentSelector
+        value={filterState.filter.instrument?.value ?? 'all'}
+      />
+      <div class="mr-2" />
+      <ZipCodeField value={filterState.filter.zipCode?.value ?? ''} />
```

Comment:
I think using `||` could be better here according to [this post](https://stackoverflow.com/questions/70334826/difference-between-and-in-javascript).
```suggestion
      <CategorySelector value={filterState.filter.category?.value || 'all'} />
      <div class="mr-2" />
      <InstrumentSelector
        value={filterState.filter.instrument?.value || 'all'}
      />
      <div class="mr-2" />
      <ZipCodeField value={filterState.filter.zipCode?.value || ''} />
```

Reply:
uhhhm this post is about the && and ?? operator... So I want the right side if the left side is null or undefined... || is for checking falsy values (including 0) but I dont really see the use case here. In fact I think we should allow empty strings as propper values as an alternative to "all". But maybe you need to elaborate more on this :D

Reply:
Nah it also shows how `||` behaves (god, JS is such a mess :D).
With `||` you can set default values a little more expressively. E.g.: At the category selector when using `??`, `false` or `0` as boolean and number values respectively are possible inputs which could lead to more potential errors i think.

```md
                        |    ||   |    &&   |    ??   |
    --------------------+---------+---------+---------+
        0, 'default'    |'default'|    0    |    0    |
       '', 'default'    |'default'|   ''    |   ''    |
    false, 'default'    |'default'|  false  |  false  |
     null, 'default'    |'default'|   null  |'default'|
undefined, 'default'    |'default'|undefined|'default'|
     true, 'default'    |  true   |'default'|  true   |
       20, 'default'    |   20    |'default'|   20    |
```